trunk/SUAVE/Analyses/Constraint_Analysis/Constraint_Analysis.py

```diff
@@ -0,0 +1,1386 @@
+                 aspect_ratio                   [Unitless]
+                 sweep_quarter_chord            [radians]
+                 aerodynamics.fuselage_factor   [Unitless]
+                              viscous_factor    [Unitless]
+
+        Outputs:
+            e          [Unitless]
+
+        Properties Used:
+
+        """  
+
+        # Unpack inputs
+        taper = self.geometry.taper
+        AR    = self.geometry.aspect_ratio
+        sweep = self.geometry.sweep_quarter_chord / Units.degrees
+        kf    = self.aerodynamics.fuselage_factor 
+        K     = self.aerodynamics.viscous_factor 
+
+        dtaper    = -0.357+0.45*np.exp(-0.0375*np.abs(sweep))
+        eff_taper = taper - dtaper
+        f_taper   = 0.0524*eff_taper**4-0.15*eff_taper**3+0.1659*eff_taper**2-0.0706*eff_taper+0.0119
+        u         = 1 / (1+f_taper*AR)
+        P         = K*cdmin
+        Q         = 1/(kf*u)
+
+        #if mach > 0.3:
+        #    keM = -0.001521*(mach/0.3-1)**10.82+1
+        #else:
+        keM = 1
+        
+        e = keM/(Q+P*np.pi*AR)
+
+
+        return e
+
+
+    def estimate_max_lift(self,highlift_type):
+        """Estimates the wing maximum lift coefficient
+
+        Assumptions:
+        None
+
+        Source:
+            D. Scholz, "Aircraft Design lecture notes", https://www.fzt.haw-hamburg.de/pers/Scholz/HOOU/AircraftDesign_5_PreliminarySizing.pdf
+
+        Inputs:
+            self.geometry.sweep_quarter_chord       [degrees]
+
+        Outputs:
+            CLmax       [Unitless]
+
+        Properties Used:
+
+        """  
+
+        # Unpack inputs
+        sweep = self.geometry.sweep_quarter_chord / Units.degrees
+
+        if highlift_type == None: # No Flaps
+            return -0.0002602 * sweep**2 -0.0008614 * sweep + 1.51    
+            
+        elif highlift_type == 'plain':  # Plain 
+            return -0.0002823 * sweep**2 -0.000141 * sweep + 1.81  
+            
+        elif highlift_type == 'single-slotted': # Single-Slotted 
+            return -0.0002599 * sweep**2 -0.002727 * sweep + 2.205      
+      
+        elif highlift_type == 'single-slotted Fowler':   # Fowler
+            return -0.000283 * sweep**2 -0.003897 * sweep + 2.501  
+      
+        elif highlift_type == 'double-slotted fixed vane':    # Double-Slotted            
+            return -0.0002574 * sweep**2 -0.007129 * sweep + 2.735  
+            
+        elif highlift_type == 'double-slotted Fowler':    # Double-slotted Fowler with Slats
+            return -0.0002953 * sweep**2 -0.006719 * sweep + 3.014 
+            
+        elif highlift_type == 'triple-slotted Fowler':    # Triple-slotted Fowler with Slats
+            return -0.0003137 * sweep**2 -0.008903 * sweep + 3.416 
+              
+        else:
+            highliftmsg = "High-lift device type must be specified in the design dictionary."
+            raise ValueError(highliftmsg)
+
+
+    def compressibility_drag(self,mach,cl):
```

Comment:
call the existing scripts since this is much the same

Reply:
Here, I am struggling with the fundamental difference between functions of SUAVE and requirements for the constraint diagram. All default functions require definitions of settings, states, etc, since thez are created for the mission segment analysis. Here, I just need 2-3 inputs for the execution with minimum input, so the constraint diagram can be initialized and executed as simple as possible. For the latest update, I still have not managed to integrate these functions into the default SUAVE

Reply:
I understand you're trying to do a simple analysis here. I'm not trying to change that. I bet you think this is frustrating because you coded this with a certain intention of this being simple.

For this function, you can easily replace it with the [original](https://github.com/suavecode/SUAVE/blob/develop/trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Drag/compressibility_drag_wing.py). Settings are not used, hence they become None. Geometry only requires wing that has a thickness to chord and sweep. State is just a dictionary that saves the outputs and provides a CL. It's not tricky to use the existing script.

We've found that having reproduced code eventually leads to bugs and just bloat. It's more confusing and it all becomes a mess eventually.